internal/service/kafka/resource_kafka_schema_registry_acl.go

```diff
@@ -0,0 +1,155 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"github.com/aiven/aiven-go-client"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var aivenKafkaSchemaRegistryACLSchema = map[string]*schema.Schema{
+	"project":      schemautil.CommonSchemaProjectReference,
+	"service_name": schemautil.CommonSchemaServiceNameReference,
+	"permission": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.StringInSlice([]string{"schema_registry_read", "schema_registry_write"}, false),
+		Description:  schemautil.Complex("Kafka Schema Registry permission to grant.").ForceNew().PossibleValues("schema_registry_read", "schema_registry_write").Build(),
+	},
+	"resource": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: schemautil.Complex("Resource name pattern for the Schema Registry ACL entry.").ForceNew().Build(),
+	},
+	"username": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: schemautil.GetACLUserValidateFunc(),
+		Description:  schemautil.Complex("Username pattern for the ACL entry.").ForceNew().Build(),
+	},
+
+	// computed
+	"acl_id": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Optional:    true,
+		Description: "Kafka Schema Registry ACL ID",
+	},
+}
+
+func ResourceKafkaSchemaRegistryACL() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The Resource Kafka Schema Registry ACL resource allows the creation and management of Schema Registry ACLs for an Aiven Kafka service.",
+		CreateContext: resourceKafkaSchemaRegistryACLCreate,
+		ReadContext:   resourceKafkaSchemaRegistryACLRead,
+		DeleteContext: resourceKafkaSchemaRegistryACLDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceKafkaSchemaRegistryACLState,
```

Comment:
It looks that we can use `schema.ImportStatePassthroughContext` instead of `resourceKafkaSchemaRegistryACLState`

Reply:
I'm not sure what this means - is it important?

Reply:
Its build in function for this, i think we can cleanup in subsequent PR maybe